molgenis-data-cache/src/main/java/org/molgenis/data/cache/l3/L3CacheRepositoryDecorator.java

```diff
@@ -53,11 +57,19 @@ public L3CacheRepositoryDecorator(Repository<Entity> decoratedRepository, L3Cach
 	@Override
 	public Stream<Entity> findAll(Query<Entity> query)
 	{
-		if (transactionInformation.isRepositoryCompletelyClean(getName()) && cacheable && query.getPageSize() > 0
-				&& query.getPageSize() <= MAX_PAGE_SIZE)
+		if (transactionInformation.isRepositoryCompletelyClean(getName()))
+		{
+			// FIXME page size for metadata is always 0, and batching is done by the postgres repository
```

Comment:
I've already added it under the header "Issues" in the FD/TD. Can also file a bug here.